lib/assets/javascripts/turbolinks.coffee

```diff
@@ -200,9 +210,26 @@ swapNodes = (targetBody, existingNodes, options) ->
         existingNode = targetNode.ownerDocument.adoptNode(existingNode)
         targetNode.parentNode.replaceChild(existingNode, targetNode)
       else
-        existingNode.parentNode.replaceChild(targetNode, existingNode)
-        onNodeRemoved(existingNode)
-        changedNodes.push(targetNode)
+        if options.append or options.prepend
+          ownerDocument = existingNode.ownerDocument
+          firstChild = existingNode.firstChild
+
+          childNodes = Array::slice.call targetNode.childNodes, 0 # a copy has to be made since the list is mutated while processing
+
+          for childNode in childNodes
+            ownerDocument.adoptNode(childNode)
```

Comment:
Although the tests pass without it, I believe this is required in some browsers as targetNode (and therefore it's childNodes) are from the parsed document tree.

The same thing is done on L210. Removing this line also allows the tests to pass.